### ISSUE TITLE
[C+ Checklist Needs Completion] [$500]  "Failed to load PDF file" message is not rendering in message preview

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#35808]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.36-1
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** @ishpaul777 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1706234706816449

## Action Performed:
 Upload a Corrupted pdf in any report

## Expected Result:
 if pdf preview fails to load "Failed to load PDF file" render

## Actual Result: 
Nothing renders

## Workaround:
unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

![PHOTO-2024-01-26-07-33-36](https://github.com/Expensify/App/assets/43996225/7f6e347b-0717-4e94-9bb3-0c6908040c8c)

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~0163673ea563ca697f</li>
        <li>Upwork Job ID: 1754473111912144896</li>
        <li>Last Price Increase: 2024-02-19</li>
<li>Automatic offers: </li>
<ul>
<li>getusha | Reviewer | 0</li>
</ul></li>
    </ul>
</details>

**Comment by ikevin127** [comment]
Can't reproduce - on my side on Android it shows an error if I try to upload a corrupted PDF.

<img width="350" alt="Screenshot 2024-02-05 at 15 30 15" src="https://github.com/Expensify/App/assets/56457735/d47679b8-1c5b-49d8-9e8f-6c5366b132c6">

**Comment by kaushiktd** [comment]
## Proposal


### Please re-state the problem that we are trying to solve in this issue.

Failed to load PDF file" message is not rendering in message preview.

### What is the root cause of that problem?

Three is a issue with mention file [here](https://github.com/Expensify/App/blob/main/src/components/Attachments/AttachmentView/AttachmentViewPdf/index.android.js#L80). It seems there is an issue with the provided file (index.android.js) where the PDF content is still being rendered even when `attachmentCarouselPagerContext` is null

### What changes do you think we should make in order to solve the problem?

To address this issue, you can modify the code to check for the null value of `attachmentCarouselPagerContext` and display an error message accordingly. Here is an example of how you can update the code:

```
{attachmentCarouselPagerContext === null ? (
    <View style={[themeStyles.flex1, themeStyles.justifyContentCenter]}>
        <Text style={props.errorLabelStyles}>{translate('attachmentView.failedToLoadPDF')}</Text>
    </View>
) : (
    // same as existing code
)}
```

### Screenshots:
![Screenshot_2024-02-06-14-40-17-67_40d2c9e2d8fb5b5aba3904834590021e](https://github.com/Expensify/App/assets/43398804/07b1f9ea-03c5-4888-a8d4-646935b9f8cd)

**Comment by ikevin127** [comment]
# Proposal

### Please re-state the problem that we are trying to solve in this issue

When successfully uploading a corrupt PDF file, the message is not being rendered in the message preview.

### What is the root cause of that problem?

> [!NOTE]
> The issue is only happening on iOS / Android native platforms.

Using this specific corrupted PDF file: [Corrupted.PDF.pdf](https://github.com/Expensify/App/files/14166298/Corrupted.PDF.pdf) makes it such that the upload is successful - we're not getting a BE error as we should with any other corrupt file.

Because the upload is successful but the file cannot be accessed on the BE side to generate its .jpg preview which is supposed to show up in the message preview, we don't see anything on native platforms but we see "Failed to load PDF file." on web.

The difference between the native / web platform is caused by the fact that on native we're applying the same styles to the Pressable that's wrapping the PDF preview (error message), as we have on web - but on native this causes the children of the Pressable to not be visible.

Both platforms show "Failed to load PDF file." from a code POV but we're only seeing it on web due to the styles.

### What changes do you think we should make in order to solve the problem?

https://github.com/Expensify/App/blob/0eb6d02b12569c42eca494d96c0c5f15eaa0b26f/src/components/PDFView/index.native.js#L188

We need to remove the styles line mentioned in the code block from the `index.native.js` file.

### Screenshots

<details>
<summary>iOS: Native</summary>

| Before | After |
| ------------- | ------------- |
| <img width="354" alt="ios-before" src="https://github.com/Expensify/App/assets/56457735/bb8bf089-b4e6-4d25-9717-b396785613a7"> | <img width="354" alt="ios-after" src="https://github.com/Expensify/App/assets/56457735/fe715157-1408-4f2b-a818-4d36619cf00f"> |

</details>

<details>
<summary>Android: Native</summary>

| Before | After |
| ------------- | ------------- |
| <img width="312" alt="android-before" src="https://github.com/Expensify/App/assets/56457735/1990d706-518f-4c42-b42c-b16c1be446c8"> | <img width="312" alt="android-after" src="https://github.com/Expensify/App/assets/56457735/29829799-258c-473d-b485-55edaba854c2"> |

</details>

**Comment by getusha** [comment]
Reviewing proposals, i will update today.

**Comment by ikevin127** [comment]
> [!NOTE]
> @kaushiktd Updated their proposal using the styles solution suggested by me after my proposal was posted - without calling it out in an **Updated proposal** comment ✌️

**Comment by kaushiktd** [comment]
@ikevin127 That was my optional solution, and unlike your idea, you are removing all styles properties, whilst I am removing only one stylesheet.

Also, my principal solution differed from yours.

**Comment by ikevin127** [comment]
1. I didn't mention exactly in my solution which style is to be removed. Obviously we're not going to remove styles which don't cause any issue - that's to be sorted later during PR, if selected.

2. Don't act like you had any idea about the styles causing the issue **before** I posted my proposal - it's clear from your RCA and solution that you didn't✌️

3. Your RCA and main solution only targets android - while the issue is reproducible on both native platforms.

**Comment by kaushiktd** [comment]
Looks like I pressed the nerve point! I knew about the styles so don't jump around to conclusions. Cintributors can see the proposals so let's leave on them. Besides I have a feeling that this issues is going to be auto close soon.

About the targeting android, I know it also happens in iOS but the issue **suggests** this is for android so I prefer to do iOS part when the same issue will be reposted in iOS, separately. If contributor has checked iOS as platform , I would have done that as well. So I would rather do it separately instead doing double job in one payment.

**Comment by getusha** [comment]
Which style property is causing the text to disappear, and any explanation why?

**Comment by ikevin127** [comment]
Looks like the `flexRow` is causing the issue here, not sure why other than the way react native styles apply between web / native platforms might be slightly different.

I checked the pressable styles together with the Failed to Load PDF wrapping View container styles and normally there shouldn't be any issue from a styles pov.

It looks like there's nothing rendered as Pressable children with that style, whereas when removing it - the children are rendered.

**Comment by kaushiktd** [comment]
@getusha Styles properties are not the main cause of this issue. When uploading a corrupted PDF file, the backend will attempt to generate an image preview for the PDF. However, if the PDF is corrupted, it won't be converted to an image and will remain in the link tag.

In the existing code, there is no context to check whether the preview is available or not on the native side. Therefore, you must check whether the preview is available or not, as mentioned in my proposal.

In addition, if you want to bypass all the check processes, you can remove the `flexrow` styles.

In current Expensify code, the `alignItemsCenter` and `flex1` styles are applied to the inner View component. here (https://github.com/Expensify/App/blob/main/src/components/PDFView/index.native.js#L148)

However, when using `flexRow` on the parent view, these styles will not behave as expected because they primarily affect the alignment of child components along the cross-axis, which is vertical in a row.

If you want to horizontally center the Text component within the row, you should either remove `flexRow` from the parent View (https://github.com/Expensify/App/blob/main/src/components/PDFView/index.native.js#L188) 
or remove `flex1` from the child View (https://github.com/Expensify/App/blob/main/src/components/PDFView/index.native.js#L145).

**Comment by getusha** [comment]
Thanks everyone for the explanation, i think it makes sense to add the condition as @kaushiktd mentioned. will continue testing and i'll update.

**Comment by ikevin127** [comment]
## Updated [proposal](https://github.com/Expensify/App/issues/35808#issuecomment-1929987589)

> i think it makes sense to add the condition as @ kaushiktd mentioned

@getusha  Not sure how you arrived to that conclusion 👀 because:

1. If we look into the code and test the proposed solution we can notice that it only "fixes"  the problem on Android.

<details>
<summary>Screenshot</summary>

<img width="801" alt="Screenshot 2024-02-11 at 22 48 37" src="https://github.com/Expensify/App/assets/56457735/6b9a573f-0958-4ac7-9a36-952b4f0314e2">

</details>

2. **Why do I quote "fixes" ?**

Because it's not an actual fix since the root cause is not understood and the solution redundantly copies a block of code which already exists within PDFView and replaces the `Content` component with that block, while the `Content` component already renders PDFView which contains that error View logic and more.

**Note**: This goes against our Contributing Guideline, precisely the DRY principle.

> [!NOTE]
> The `Content` that's returned when `attachmentCarouselPagerContext === null` renders the `BaseAttachmentViewPdf` component which in turn renders the actual PDFView which already contains the error View code block when there's an error + other logic which should not be discarded.

**Again**: this is a simple styling conflict issue, as mentioned in my proposal - has nothing to do with the error View not being rendered - it is rendered, it's just not visible because of the styles. See the updated [proposal](https://github.com/Expensify/App/issues/35808#issuecomment-1929987589) for more details.

I think somebody from Expensify should take a look at the current state of this issue because with statements like:

> About the targeting android, I know it also happens in iOS but the issue suggests this is for android so I prefer to do iOS part when the same issue will be reposted in iOS, separately. If contributor has checked iOS as platform , I would have done that as well. So I would rather do it separately instead doing double job in one payment.

especially that last sentence, I don't like where this is going 😅

**Comment by kaushiktd** [comment]
> **Note**: This goes against our Contributing Guideline, precisely the DRY principle.

Consider me a friend while I say this, but don't push this, my dear fellow! The way you phrase this sounds like you're accusing them of not knowing what they're doing. They are "assignees" for a reason.

Also, it makes no difference to me or the contributors where my last sentences end because I had a conversation on Slack when I was in a similar circumstance. Since they discontinued bug reporting and made it purely internal, there is no purpose in registering the identical bug on iOS for this. They are coders, so if they believe this should also be done in iOS, they will let us know in a separate job. As of now only ANDROD is check marked. It is common sense for me that any bug reported for android, I should check them in iOS. I knew that. Stop assuming they are ignoring your suggestion or favoring mine and making such long comments. They inspect everything. Just because they do not put their views about every proposal, does not mean they do not check each comment here.

**Comment by ikevin127** [comment]
> don't push this

Don't push what exactly ? If you don't mind me asking.
We're not friends nor dear fellows - I don't know you personally 🙂, all I know about you is that you are trying really hard to push redundant code to our codebase while not fully understanding the root cause of the issue.

Just because whomever reported this bug did not check on iOS, that doesn't mean we should completely ignore the fact that this happens on iOS and disregard the guidelines for a quick pay, regardless of making a dent within the codebase.

> Stop assuming they are ignoring your suggestion or favoring mine and making such long comments. They inspect everything. Just because they do not put their views about every proposal, does not mean they do not check each comment here. Trust the process. They will make right decision.

That's partly true and shouldn't always be taken as absolute truth as based on recent events I can say that that's not always the case. Take [this issue](https://github.com/Expensify/App/issues/35207) as an example.

As you can see from the mentioned issue, the reviewer already made their decision but upon further review from an Expensify team member, the decision was to go with the correct RCA and solution which was best for the codebase and I think we're dealing with the same situation here.

We have to acknowledge that sometimes the reviewer doesn't have as deep of an understanding of an issue as a Contributor that spend 1-2 hours debugging that issue has when it comes to the code involved. That's completely fine, but that doesn't mean I won't do anything in my power to bring that to their attention - especially when it comes to keeping the codebase clean.

I understand that some Contributors might have other motivations than keeping the codebase clean and adhering to the guidelines, but this does not mean they have the right to silence somebody who sees things differently ✌

I'm going to stop the back and forth here as given all the context provided I think it's enough for the reviewer / team to make their decision 🚀

**Comment by kaushiktd** [comment]
Again long msg and you were asking for pushing what!! I said "Consider" but nvm. Great talk and good luck!🙂

**Comment by getusha** [comment]
thanks for the detailed RCA @ikevin127, will test your proposal too. i appreciate the patience.

**Comment by getusha** [comment]
@kaushiktd i found your proposal to be quite incomplete and a bit hard to test, could you add a bit more details on the implementation based on the component being class based currently?

**Comment by kaushiktd** [comment]
Sure, @getusha

**Comment by kaushiktd** [comment]
I have a question. In my proposal I have already given detailed explanation on class components for android. Are you requesting to do this for iOS as well? @getusha

**Comment by getusha** [comment]
@kaushiktd my bad i applied your solution at the wrong place.

@ikevin127 this is the diff

```diff
diff --git a/src/components/PDFView/index.native.js b/src/components/PDFView/index.native.js
index bfdb80131a..af0d64431b 100644
--- a/src/components/PDFView/index.native.js
+++ b/src/components/PDFView/index.native.js
@@ -142,7 +142,7 @@ class PDFView extends Component {
         const containerStyles =
             this.state.shouldRequestPassword && this.props.isSmallScreenWidth
                 ? [this.props.themeStyles.w100, this.props.themeStyles.flex1]
-                : [this.props.themeStyles.alignItemsCenter, this.props.themeStyles.flex1];
+                : [themeStyles.w100, themeStyles.h100];
 
         return (
             <View style={containerStyles}>
```

**Comment by ikevin127** [comment]
@getusha That does not make any sense, can you make sure you pulled latest main and rebuild your android app ?

<details>
<summary>Android: step by step with latest main and fresh build</summary>

https://github.com/Expensify/App/assets/56457735/6c9466ea-1ba1-40e5-8d1c-7dfb1398cc55

</details>

Looks like there are some differences in your diff as the component changed from class to functional recently.
I don't understand, does nothing change on your side when you save the file with the suggested changes ? 👀
Can you please ask in the C+ channel if anyone else can apply the changes and test this ?

**Comment by kaushiktd** [comment]
> @kaushiktd my bad i applied your solution at the wrong place.

So can I assume I do not have to do anything? @getusha

**Comment by getusha** [comment]
@ikevin127 looks like rebuilding the android app make it work.

>So can I assume I do not have to do anything? @getusha

@kaushiktd yes

**Comment by ikevin127** [comment]
It happens sometimes for the apps to use cached old code and have weird bugs that cannot be explained if a rebuild wasn't performed in a while after many changes were pulled into the codebase ♻ I know that building on Android can be a pain 😪

Glad you managed to get it work and confirm the solution as working 👍

**Comment by getusha** [comment]
@kaushiktd your solution will cause a regression when you try to upload a new PDF it will show the error message no matter what.

![Screenshot 2024-02-18 at 6 49 40 PM](https://github.com/Expensify/App/assets/75031127/0f00ea20-77c2-43b2-b33f-95cc985e7493)

**Comment by getusha** [comment]
@ikevin127 this is what i am seeing when i try to upload the corrupt pdf.

![Screenshot 2024-02-18 at 6 50 42 PM](https://github.com/Expensify/App/assets/75031127/16956f12-f62f-4c92-81c0-85ebe86b9d47)

**Comment by getusha** [comment]
@kaushiktd's alternative solution seems to work, will test it further.

**Comment by ikevin127** [comment]
@getusha Hmm, weird - it looks good on my side 🧐

Most likely because of the **h100**, we just need to pass height: min-content or something similar that would only give it as much height as the view has.

Will get back and let you know exactly - but I think since the solution is correct (a styling issue) we can work this out during the PR, what so you think ?

**Comment by getusha** [comment]
>but I think since the solution is correct (a styling issue) we can work this out during the PR, what do you think ?

I agree your RCA is correct and you're the first to point where the issue lies (styling). but it could create complications if we end up using @kaushiktd's alternative solution in the PR. imo its better to clear that up here.

**Comment by ikevin127** [comment]
> @ kaushiktd's alternative solution seems to work, will test it further.

@getusha That was my main solution before I updated my proposal - which was added as alternative solution **AFTER** I posted my initial proposal.

Please check comments edit history ✌️

**Comment by kaushiktd** [comment]
> but it could create complications if we end up using @kaushiktd's alternative solution in the PR.

Not sure what complications you are referring to here but I tested again and it works perfectly! 
https://drive.google.com/file/d/14mhmdLmoSGjlUAjFqkmwFnpFlyr7iyTS/view

Also I belive I provided the proposal, and then updated it multiple times. Unfortunately from my side I can not check exactly when was my proposal updated so I will not be pushy here.

Also, it doesn't imply I'm incorrect if I don't respond to every comment. I don't tend to be pushy or aggressive because I have complete faith in the assignee. Although I oppose splitting the amount, I will accept the assignee's decision in this regard.

**Comment by getusha** [comment]
>Not sure what complications you are referring to here but I tested again and it works perfectly!

@kaushiktd did you test your main solution or alternative?

**Comment by kaushiktd** [comment]
> @kaushiktd did you test your main solution or alternative?

Made that video after testing both solutions, main and alternative.

**Comment by getusha** [comment]
>@getusha I made that video after testing both solutions, main and alternative.

I double confirmed unfortunately the main solution causes the issue i mentioned above.

**Comment by kaushiktd** [comment]
Ishpaul777's comment just caught my attention and as per that in latest code merge this issue may have been fixed. Have you checked that?

**Comment by ikevin127** [comment]
## Updated [proposal](https://github.com/Expensify/App/issues/35808#issuecomment-1929987589) (second time)
- final update of proposal based on the latest context as the last update would create regression ([#35808 (comment)](https://github.com/Expensify/App/issues/35808#issuecomment-1951368216))
- this update takes care of both scenarios (chat comment and attachment modal open with "Failed to load PDF file" error)

cc @getusha 
## ___

> Ishpaul777's comment just caught my attention and as per that in latest code merge this issue may have been fixed. Have you checked that?

### I can confirm that the issue is still present with latest main branch.

> Also I belive I provided the proposal, and then updated it multiple times. Unfortunately from my side I can not check exactly when was my proposal updated and in my proposal I have made everything clear so far. so I will not be pushy here.
> 
> Also, it doesn't imply I'm incorrect if I don't respond to every comment. I don't tend to be pushy or aggressive because I have complete faith in the assignee. Although I oppose splitting the amount, I will accept the assignee's decision in this regard.

This is not being aggressive or pushy as you keep insinuating, this is about making sure the reviewer is aware of the proposal edit history and who came with the correct RCA / solution **first** - you don't have to make this a personal attack as this is an argument based on real data (timestamps) which can be verified by anyone by hovering the cursor over each edit timestamp (see videos below).

So here we go, below you can find timestamps / snapshoots of the proposals and their edits in **chronological** order to avoid any confusion when it comes to the reviewing / selection process:

<details>
<summary>Proposals edit history (spoiler)</summary>

 @kaushiktd Feb 6, 11:24 AM - Initial proposal
- watching the video we can see that there's no indication of a styling issue in either RCA or solution
<details>
<summary>Video of proposal</summary>

https://github.com/Expensify/App/assets/56457735/68b73cb6-0aed-4770-a857-94c313a096e5

</details>
@kaushiktd Between Feb 6, 11:24 AM - Feb 6, 2:30 PM - 2 minor edits adding to the initial RCA / solution

- no mention of the styling issue yet

@ikevin127 Feb 6, 5:06 PM - Initial proposal
- first mention of the correct RCA / solution according to reviewer's comment ([#35808 (comment)](https://github.com/Expensify/App/issues/35808#issuecomment-1951370433))
<details>
<summary>Video of proposal</summary>

https://github.com/Expensify/App/assets/56457735/1f8b7482-cf6e-4d3f-ae65-0cba370ac0de

</details>

 @kaushiktd Feb 7, 07:13 AM - 3rd edit of proposal, a day later after my initial proposal
- watching the video we can see that the alternative was added a day after my initial proposal was posted, I will let the reviewers decide if this was inspired by mine or not 
- added the alternative quickly with permalink broken (line doesn't exist in that file - check video below)
<details>
<summary>Video of edited proposal</summary>

https://github.com/Expensify/App/assets/56457735/27b9cc3e-538f-4b9c-add7-0a9fd7c2238e

</details>

 @kaushiktd Feb 7, 07:16 AM - 4rd edit of proposal, specified exactly what style should be removed but not from what file
- permalink to the referenced code line still broken (line doesn't exist in that file - check video below)
<details>
<summary>Video of edited proposal</summary>

https://github.com/Expensify/App/assets/56457735/282829b7-1cd4-4505-823c-305f8643aab0

</details>

</details>

This is all I had to say with regards to the edit timestamps and I hope this will be taken into account when assigning the Contributor. In any case, this is the proof to support my case for the 50/50 split in case the RCA / solution I identified and proposed initially will be used by another Contributor 🙏

**Comment by getusha** [comment]
thanks @ikevin127 conditionally removing that styling makes sense! i'll test it a bit more and provide an update.
i think splitting the bounty is fair, what do you guys think? @kaushiktd

**Comment by kaushiktd** [comment]
@getusha I will respect your decision without hesitation. 🙂

**Comment by ikevin127** [comment]
@getusha I disagree splitting the bounty if I'm assigned here, since based on the info provided in https://github.com/Expensify/App/issues/35808#issuecomment-1952482113 (spoiler) is clear at least from my side who came up with the correct RCA / solution first and who edited their proposal using that info 1 day later:
- without explaining RCA
- without posting a **Proposal updated** comment

having no idea of the correct RCA / solution in their initial proposal.

Just so we're clear here - I'm not going to do the work myself if we're going to split.

The only way we would split here, **which I still think is unfair** (no basis), would be for somebody else to do the work using my proposal.

If you disagree please support your case for a split using the proposal edits history - I'm curious how you came to that suggestion 👀

Otherwise we should have somebody else review this split situation once assigned, hopefully it will clear things up!

**Comment by getusha** [comment]
@ikevin127 the alternative solution @kaushiktd was the solution that was working without any regressions (removing a specific styling without having to remove all), and if one provides a clear RCA and a solution that'll cause a regression i don't think we should be selecting that. and splitting is fair because one provided the RCA one has a working solution.

**Comment by ikevin127** [comment]
@getusha I think there's a misunderstanding here, since my initial proposal (unedited) identified the correct solution just like the alternative of the second proposal did, which was added 1 day later without mentioning the update and giving any credit.

> splitting is fair because one provided the RCA one has a working solution

I disagree, I think I provided both, then the other proposal just inspired their alternative from my main solution without even explaining it, plus having a broken permalink which could mean removing that style from any other part of the component 😶‍🌫️

You can't say that one provided a working solution (without any explanation) when the code line from that solution was identified by somebody else 1 day before 👎

But you seem very set for a bounty split and I'm not going to keep this back and forth forever so I think somebody else should look into this before final assignment and decide if this is even worth entertaining a bounty split.

> [!NOTE]
>  The only way we would split here, which I still think is unfair (no basis), would be for somebody else to do the PR work using my latest proposal.

**Comment by getusha** [comment]
>We need to remove the styles line mentioned in the code block from the index.native.js file.

Your initial [solution](https://github.com/Expensify/App/issues/35808#issuecomment-1929987589) was to remove the entire `styles` line, which i oppose as it might cause other issues. Then, @kaushiktd proceeded to update the proposal, indicating which styling to remove, not the whole line.  Then, you changed your solution. This solution had some issues during testing, and the recent update includes removing the specific style @kaushiktd [proposed](https://github.com/Expensify/App/issues/35808#issuecomment-1929098883). 

>We can also remove the flexRow style from the wrapping Pressable but this is less preferred since the Pressable wraps the rest of the PDFView logic, not only the error View and we shouldn't mess with these styles since they are the same on web.

^ was added after @kaushiktd's alternative proposal.

the only reason i proposed splitting the bounty is only because the RCA @kaushiktd provided is wrong but the alternative solution works well without any issues, and i suggest you to be open to working as a team.  and there is no need to argue here, will let internal engineer decide. 

🎀 👀 🎀

**Comment by ikevin127** [comment]
> Your initial https://github.com/Expensify/App/issues/35808#issuecomment-1929987589 was to remove the entire styles line, which i oppose as it might cause other issues

Correct, no issues here!

> @ kaushiktd proceeded to update the proposal, indicating which styling to remove, not the whole line.

Correct but where's the credit ?
Are we just acting like that idea just came magically overnight out of nowhere ? 😅 This is unbelievable!

Plus where's the proposal updated comment ? I thought we discard major proposal changes / updates if the comment is not posted as per [Propose a solution for the job](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#propose-a-solution-for-the-job), rule 6, dot 3:

> If you want to make an entirely new proposal or update an existing proposal, please go back and edit your original proposal, then post a new comment to the issue in this format to alert everyone that it has been updated

.

> RCA @ kaushiktd provided is wrong but the alternative solution works well without any issues

That might be the case but then still that was not selected in the end but the one I suggested in my latest [proposal update](https://github.com/Expensify/App/issues/35808#issuecomment-1929987589) as per comment:

> thanks @ikevin127 conditionally removing that styling makes sense!

.

> i suggest you to be open to working as a team

Oh, I'm extremely open to working as a team and I don't consider this as arguing but supporting my case 🙏

I just don't think we have the same definition of "working as a team" because what **kaushiktd** did here I don't consider team work: using somebody else's work as a Hail Mary to still get some benefit from the issue when realising their initial proposal was wrong, that's not team work in my book.

**Comment by kaushiktd** [comment]
@getusha I was against splitting the bounty because I didn't want to make this platform a fighting place. But my friend here being aggressively pushy while denying of being one, ever since the decision was made in my favour. He even indirectly mentioned assignee doesn't know their job but now this guy is breaking the limits. The ideal decision as per the rules should be full payment to me. Even I've faced the same situation tons of time here when someone's proposal got selected that had some copied content from my proposal but it was way better than mine with their own proposal, and I had to accept it without opposing. I should be paid full as I didn't break any rule here but again, I'll let the decision sit in assignee's capable hands without letting myself emotionally incapable here. This will be the last comment from me in this regard.

**Comment by ikevin127** [comment]
@kaushiktd Wow, you're unbelievable! 🤣

> He even indirectly mentioned assignee doesn't know their job but now this guy is breaking the limits.

This is what I said (https://github.com/Expensify/App/issues/35808#issuecomment-1937885115):

> We have to acknowledge that sometimes the reviewer doesn't have as deep of an understanding of an issue as a Contributor that spend 1-2 hours debugging that issue has when it comes to the code involved. That's completely fine, but that doesn't mean I won't do anything in my power to bring that to their attention - especially when it comes to keeping the codebase clean.

I made my case as to why the initial reviewer's decision about your initial proposal is wrong as the RCA / solution is incorrect for the reasons detailed there in the same comment -> which the reviewer confirmed later here https://github.com/Expensify/App/issues/35808#issuecomment-1951367711.

> I should be paid full as I didn't break any rule here

Wow, just wow - no words 😅

I think the only way to make sense of what you said in your last comment might be to assume there's a language barrier since you're throwing out some wild accusations based on not understanding what was written and the meaning of these sentences.

If you want half of the bounty you should do some work and handle the PR here, otherwise leave it on me, admit you were wrong and look for other issues to contribute to ✌

**Comment by jasperhuangg** [comment]
Hey guys, I've gone over what has happened. At this point, @kaushiktd's proposed alternative solution was the first to work with no regressions. However, they weren't able to get to that point without @ikevin127's RCA and direction.

So we're going to:
- Assign @kaushiktd to work on the issue, since they provided the working solution
- Acknowledge that this wouldn't have been possible without the ideas @ikevin127 put forth in their proposals, and split the bounty evenly between @kaushiktd and @ikevin127 

Thank you both for your patience with this matter and for your enthusiasm with solving the bug!

In the future, let's keep in mind that everyone is here to help with solving bugs, and these types of opportunities wouldn't be possible without the collaborative and supportive community that you're both a part of.

**Comment by jasperhuangg** [comment]
cc ^ @greg-schroeder

**Comment by kaushiktd** [comment]
@getusha @jasperhuangg Can one of you please send upwork offer manually?

**Comment by ikevin127** [comment]
If that's really needed now, then we should both have offers sent for $250 each.

Otherwise I wouldn't worry about the payment now as that can be sorted 7 days after the fix is deployed to production - when hold for payment is due.

**Comment by kaushiktd** [comment]
> @getusha @jasperhuangg Can one of you please send upwork offer manually?

Any update on this?

**Comment by getusha** [comment]
@kaushiktd there is no need to worry, offer can be sent after PR is merged and regression period passes.

**Comment by greg-schroeder** [comment]
Reviewing

**Comment by greg-schroeder** [comment]
Yeah, I mean we're still not even in the payment period yet, but I'll send an offer to both just so it's ready right away

**Comment by ikevin127** [comment]
@greg-schroeder Friendly bump for payments 🫡

**Comment by greg-schroeder** [comment]
I was OOO yesterday, but I'll process this today

**Comment by greg-schroeder** [comment]
All payments complete. @getusha can you complete [the checklist](https://github.com/Expensify/App/issues/35808#issuecomment-1972141716) so we can close this out? Thanks!

**Comment by getusha** [comment]
BugZero Checklist: The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

 [@getusha] The PR that introduced the bug has been identified. Link to the PR: https://github.com/Expensify/App/pull/21714
 [@getusha] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment: https://github.com/Expensify/App/pull/21714/files#r1519516670
 [@getusha] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion: N/a
 [@getusha] Determine if we should create a regression test for this bug. **Yes**
 [@getusha] If we decide to create a [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) for the bug, please propose the regression test steps to ensure the same bug will not reach production again.

**Regression Test Proposal**

1. Open any chat
2. Add attachment
3. Send corrupted pdf (https://github.com/Expensify/App/files/14166298/Corrupted.PDF.pdf)
4. Verify that `Failed to load PDF file` renders correctly.

Do we agree 👍 or 👎

**Comment by greg-schroeder** [comment]
Filing regression test and closing!
